spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2017 Boundless, http://boundlessgeo.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License
+ */
+package com.boundlessgeo.spatialconnect.services.authService;
+
+import android.content.Context;
+import android.util.Base64;
+import android.util.Log;
+
+import com.boundlessgeo.spatialconnect.scutilities.HttpHandler;
+import com.github.rtoshiro.secure.SecureSharedPreferences;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Locale;
+
+import okhttp3.Response;
+
+public class SCExchangeAuthMethod implements ISCAuth {
+
+    private static String LOG_TAG = SCExchangeAuthMethod.class.getSimpleName();
+    private static final String USERNAME = "username";
+    private static final String PWD = "pwd";
+    private static final String ACCESS_TOKEN = "accessToken";
+    private SecureSharedPreferences settings;
+    private String serverUrl;
+    private Context context;
+    private String clientId;
+
+    public SCExchangeAuthMethod(Context context, String serverUrl, String clientId) {
+        this.context = context;
+        this.settings = new SecureSharedPreferences(context);
+        this.serverUrl = serverUrl;
+        this.clientId = clientId;
+    }
+
+    @Override
+    public boolean authFromCache() {
+        String u = username();
+        String p = getPassword();
+        if (u != null && p != null) {
+            return authenticate(u,p);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean authenticate(String username, String pwd) {
+        return auth(username, pwd);
+    }
+
+    @Override
+    public boolean refreshToken() {
```

Comment:
Is this ever called anywhere?  I couldn't find it.

Reply:
Not yet, the next PR will actually call it,  will need to check everytime it attempts to auth against the cache, and also need to figure out a good way to check on every http request, there could be an instance when the app is open and the token has expired.